feat: add CLI technical specification

yahgw-ai opened the pull request:
## Summary
- Add comprehensive technical specification for the AEP Fee Calculator CLI
- Defines command-line interface with --rpc-url (required) and optional date/directory parameters
- Specifies simple orchestration of existing components in dependency order

no-issue: adding CLI specification as requested

## Test plan
- [ ] Review specification for completeness and accuracy
- [ ] Verify specification aligns with existing component interfaces
- [ ] Confirm execution order matches system dependencies

yahgw-ai committed:
feat: add CLI technical specification

yahgwai commented on docs/specs/cli.md:
```diff
@@ -0,0 +1,244 @@
+
+### Infrastructure Requirements
+
+- Access to Arbitrum Nova archive RPC endpoint
```
No required to be nova, just needs an rpc connection of some kind, provided by command line args

yahgwai commented on docs/specs/cli.md:
```diff
@@ -0,0 +1,244 @@
+       ↓
+   Balance Fetcher
+       ↓
+   Event Scanner
```
This is called a recipient recieved scanner now

yahgwai commented on docs/specs/cli.md:
```diff
@@ -0,0 +1,244 @@
+
+### Integration Points
+
+- Environment variables: RPC_URL, CHAIN_ID, STORE_DIR
```
Do not need a chain id, this is always fetched from the rpc provider.

yahgwai replied:
The env vars should mirror the command line args

yahgwai commented on docs/specs/cli.md:
```diff
@@ -0,0 +1,244 @@
+
+- Environment variables: RPC_URL, CHAIN_ID, STORE_DIR
+- File system: Shared store directory for inter-component data exchange
+- Process execution: Child process spawning for each component
```
No, they all run in the same process. Remove this point completely.

yahgwai commented on docs/specs/cli.md:
```diff
@@ -0,0 +1,244 @@
+2. **Environment Setup**
+
+   - Set RPC_URL environment variable from --rpc-url
+   - Set CHAIN_ID to 42170 (Arbitrum Nova)
```
Never nova - we never hard code a chain id

yahgwai commented on docs/specs/cli.md:
```diff
@@ -0,0 +1,244 @@
+
+2. **Environment Setup**
+
+   - Set RPC_URL environment variable from --rpc-url
```
Parameters are either set from the comman line or through environment variables by the user, not by us. Parameters can be set either. The parameters are then passed directly to components, since all components run in the same process.

yahgwai commented on docs/specs/cli.md:
```diff
@@ -0,0 +1,244 @@
+3. **Component Execution**
+
+   - For each component in sequence:
+     - Spawn child process executing the component
```
Not a child process - remove all references to that in this spec

yahgwai commented on docs/specs/cli.md:
````diff
@@ -0,0 +1,244 @@
+```typescript
+interface EnvironmentConfig {
+  RPC_URL: string; // From --rpc-url
+  CHAIN_ID: "42170"; // Hard-coded for Arbitrum Nova
````
Not necessary

yahgw-ai committed:
fix: address PR comments on CLI spec

- Remove hard-coded CHAIN_ID references (chain ID fetched from RPC)
- Update to clarify components run in same process, not child processes
- Make spec chain-agnostic (not Nova-specific)
- Clarify configuration can come from CLI args or environment variables
- Update Event Scanner to Recipient Recieved Scanner throughout

yahgwai commented on docs/specs/cli.md:
```diff
@@ -143,15 +142,14 @@ interface CLIArguments {
-  START_DATE?: string; // From --start-date
-  END_DATE?: string; // From --end-date
+interface ComponentConfig {
+  RPC_URL: string; // From --rpc-url or environment
```
No, these are only from the environment.

yahgw-ai committed:
fix: remove integration testing section from CLI spec

yahgw-ai committed:
fix: clarify components only read from environment variables

- CLI sets environment variables from command-line arguments
- Components read configuration only from environment variables
- Remove ambiguity about configuration sources

yahgwai commented on docs/specs/cli.md:
```diff
@@ -142,14 +143,14 @@ interface CLIArguments {
-  STORE_DIR?: string; // From --store-dir or environment or default
-  START_DATE?: string; // From --start-date or environment
-  END_DATE?: string; // From --end-date or environment
+interface EnvironmentVariables {
```
remove these

yahgw-ai committed:
fix: remove environment variables interface per review

yahgw-ai committed:
fix: remove environment setup section per review

yahgw-ai committed:
fix: remove all remaining environment variable references
